Dockerfile

```diff
@@ -25,32 +25,50 @@
 #     -i -t --rm \
 #     --runtime=nvidia \
 #     core3d/danesfield \
-#     danesfield/tools/material_classifier.py --cuda ...
+#     danesfield/tools/material_classifier.py --cuda ...    
+
+# Download and install miniconda3
+# Based on https://github.com/ContinuumIO/docker-images/blob/fd4cd9b/miniconda3/Dockerfile
+# The step to update 'conda' is necessary to avoid the following error when
+# downloading packages (see https://github.com/conda/conda/issues/6811):
+#
+#     IsADirectoryError(21, 'Is a directory')
+#
+
+FROM nvidia/cuda:10.0-devel-ubuntu18.04
 
-FROM nvidia/cuda:9.0-runtime-ubuntu16.04
 LABEL maintainer="Max Smolens <max.smolens@kitware.com>"
 
+COPY ply2txt.py ply2txt.py
+
+COPY patches /patches
+
 # Install prerequisites
-RUN apt-get update && apt-get install -y --no-install-recommends \
-    bzip2 \
+RUN apt-get update && apt-get install -y software-properties-common && apt-get update && \
+  add-apt-repository -y ppa:ubuntu-toolchain-r/test && add-apt-repository -y ppa:ubuntugis/ppa && apt-get update && \
+  DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  sudo \
+  make \
+  git \
+  bzip2 \
     ca-certificates \
     curl \
     libgl1-mesa-glx \
     libglu1-mesa \
     libxt6 \
     xvfb \
-    sudo && \
+    gdal-bin \
```

Comment:
Do Colmap and / or VisSat require GDAL installed at the system level?  We install GDAL via Conda at least for the Python components so we shouldn't need to apt-install it just for those.